### DISCUSSION TITLE
JSON Input type / GraphQl Editor

### DISCUSSION THREAD

**Discussion by octoro-software** [umbraco/Umbraco-CMS#12898] [Features and ideas]
Hey, as part of my project i use graphQL to write queries for different apps and would be great if we could get a JSON editor or GraphQL Editor to write queries. The output would be a string of the query. Here is what im currently doing using a text area.

![Untitled](https://user-images.githubusercontent.com/61507338/186508686-c8cdc2c2-72e6-4921-8dff-85483919159d.png)

**Comment by leekelleher** [comment]
@octoro-software If you need something in the meantime, my Contentment package has a Code Editor that can be configured for JSON input/syntax.
https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/code-editor.md

But don't let this distract from a wider conversation about a JSON editor in Umbraco CMS itself.

**Reply by octoro-software** [reply]
@leekelleher I have came across this. Problem with contentment in heartcore is I would need to create 100's of data types for all the different queries where as i can just use the content editor here and adjust according to the project. Conentment is great for me for static resources such as a list of icon definitions for example.

**Reply by leekelleher** [reply]
Ah, if this is Umbraco Heartcore then ignore me, as only Data List editor is available from Contentment on Heartcore.

Just to clarify if there any confusion about Contentment, the package contains several editors, not just the Data List one, so with the Code Editor, I'd expect that'd be a drop-in replacement to the textarea editor, (without the need to create 100s of data types). But, alas, not on Heartcore.
